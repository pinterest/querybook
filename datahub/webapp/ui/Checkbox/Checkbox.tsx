import React from 'react';
import classNames from 'classnames';
import './Checkbox.scss';

export interface ICheckboxProps {
    className?: string;
    disabled?: boolean;

    title?: string;
    value?: boolean;
    onChange?: (value: boolean) => any;
}

export const Checkbox: React.FunctionComponent<ICheckboxProps> = ({
    className = '',
    disabled,

    title = '',
    value,
    onChange,
    children,
}) => {
    const checkboxClass = classNames({
        [className]: Boolean(className),
        checked: value,
    });

    return (
        <div className="Checkbox">
            <label
                className={checkboxClass}
                onClick={
                    disabled
                        ? null
                        : (event) => {
                              if (onChange) {
                                  onChange(!value);
                                  event.stopPropagation();
                                  event.preventDefault();
                              }
                          }
                }
            >
                <input
                    type="checkbox"
                    disabled={disabled}
                    checked={value}
                    readOnly
                />{' '}
                {title}
                {children}
            </label>
        </div>
    );
};
