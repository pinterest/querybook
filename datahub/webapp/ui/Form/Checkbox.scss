.Checkbox {
    label {
        line-height: 1;
        display: flex;
        flex-direction: row;
        white-space: nowrap;
        position: relative;

        input[type='checkbox'] {
            cursor: pointer;
            appearance: unset;
            border: solid 1px var(--border-color);
            background-color: var(--bg-color);

            margin-right: 8px;
            width: 16px;
            height: 16px;
            outline: none;
        }

        &.checked::before {
            content: ' ';
            background-color: var(--border-color);
            width: 12px;
            height: 12px;
            display: block;
            margin: 2px 0px 0px 2px;
            border-radius: 2px;

            position: absolute;
            left: 1px;
            top: 1px;
        }
    }
}
