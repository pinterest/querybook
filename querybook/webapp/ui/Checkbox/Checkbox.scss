.Checkbox {
    label {
        line-height: 1;
        white-space: nowrap;
        position: relative;

        display: flex;
        flex-direction: row;
        align-items: center;

        cursor: pointer;
        user-select: none;

        input[type='checkbox'] {
            appearance: unset;
            border: solid 1px var(--border-color);
            background-color: var(--bg-color);
            cursor: pointer;

            margin-right: 8px;
            width: 16px;
            height: 16px;
            outline: none;
        }

        &.checked::before {
            content: ' ';
            background-color: var(--border-color);
            width: 12px;
            height: 12px;
            display: block;
            margin: 2px 0px 0px 2px;
            border-radius: 2px;
            cursor: pointer;
            pointer-events: none;

            position: absolute;
            left: 1px;
            top: 1px;
        }
    }
}
