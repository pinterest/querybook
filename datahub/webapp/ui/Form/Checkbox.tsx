import React from 'react';
import classNames from 'classnames';
import './Checkbox.scss';

export interface ICheckboxProps {
    className?: string;
    disabled?: boolean;

    title?: string;
    value?: boolean;
    onChange?: (value: boolean) => any;
}

export const Checkbox: React.FunctionComponent<ICheckboxProps> = ({
    className = '',
    disabled,

    title = '',
    value,
    onChange,
    children,
}) => {
    const checkboxClass = classNames({
        [className]: Boolean(className),
        checked: value,
    });

    return (
        <div className="Checkbox">
            <label className={checkboxClass} disabled={disabled}>
                <input
                    type="checkbox"
                    disabled={disabled}
                    checked={value}
                    onChange={(event) => {
                        if (onChange) {
                            onChange(event.target.checked);
                            event.stopPropagation();
                        }
                    }}
                />{' '}
                {title}
                {children}
            </label>
        </div>
    );
};
