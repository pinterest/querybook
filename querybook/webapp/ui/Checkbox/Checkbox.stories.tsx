import React, { useState } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Checkbox } from './Checkbox';

export default {
    title: 'Form/Checkbox',
    decorators: [centered],
};
export const _Checkbox = (args) => {
    const [state, setState] = useState(false);

    return (
        <>
            <Checkbox {...args} onChange={setState} value={state} />
        </>
    );
};

_Checkbox.args = {
    title: 'checkbox',
    disabled: false,
};
