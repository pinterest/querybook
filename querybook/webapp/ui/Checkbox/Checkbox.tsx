import React from 'react';
import clsx from 'clsx';
import './Checkbox.scss';

export interface ICheckboxProps {
    className?: string;
    disabled?: boolean;

    title?: string;
    value?: boolean;
    onChange?: (value: boolean) => any;
}

export const Checkbox: React.FunctionComponent<ICheckboxProps> = ({
    className = '',
    disabled,

    title = '',
    value,
    onChange,
    children,
}) => {
    const checkboxClass = clsx({
        [className]: Boolean(className),
        checked: value,
    });

    return (
        <div className="Checkbox">
            <label
                className={checkboxClass}
                onClick={
                    disabled
                        ? null
                        : (event) => {
                              if (onChange) {
                                  onChange(!value);
                                  event.stopPropagation();
                                  event.preventDefault();
                              }
                          }
                }
            >
                <input
                    type="checkbox"
                    disabled={disabled}
                    checked={value}
                    readOnly
                />{' '}
                {title}
                {children}
            </label>
        </div>
    );
};
