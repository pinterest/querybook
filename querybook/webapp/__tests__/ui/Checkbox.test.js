import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Checkbox } from '../../ui/Checkbox/Checkbox';

it('renders without crashing', () => {
    shallow(<Checkbox />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Checkbox />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
