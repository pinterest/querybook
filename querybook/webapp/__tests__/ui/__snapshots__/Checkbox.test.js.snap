// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="Checkbox"
>
  <label
    className=""
    onClick={[Function]}
  >
    <input
      readOnly={true}
      type="checkbox"
    />
     
  </label>
</div>
`;
